src/huggingface_hub/inference/_mcp/cli.py

```diff
@@ -68,6 +70,60 @@ def _sigint_handler() -> None:
             # Windows (or any loop that doesn't support it) : fall back to sync
             signal.signal(signal.SIGINT, lambda *_: _sigint_handler())
 
+        # Handle inputs (i.e. env variables injection)
+        if len(inputs) > 0:
+            print(
+                "[bold blue]Some initial inputs are required by the agent. "
+                "Please provide a value or leave empty to load from env.[/bold blue]"
+            )
+            for input_item in inputs:
+                input_id = input_item["id"]
+                description = input_item["description"]
+                env_special_value = "${input:" + input_id + "}"  # Special value to indicate env variable injection
+
+                # Check env variables that will use this input
+                input_vars = list(
+                    {
+                        key
+                        for server in servers
+                        for key, value in server.get("config", {}).get("env", {}).items()
+                        if value == env_special_value
```

Comment:
potentially, you should also check `config.headers`: for HTTP or SSE servers it's expected most keys will end up in headers.

see related doc PR from me here: https://github.com/microsoft/vscode-docs/pull/8437

Reply:
i've pushed something on your JS PR actually:
https://github.com/huggingface/huggingface.js/pull/1501/commits/d154e3e62be7036e5474b002c92d54f1e18b41b5

(see the comments there..)